src/Compilers/Core/Portable/InternalUtilities/SharedStopwatch.cs

```diff
@@ -2,15 +2,23 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Diagnostics;
+using System.Runtime.InteropServices;
 
 namespace Roslyn.Utilities
 {
     internal readonly struct SharedStopwatch
     {
+        static SharedStopwatch()
+        {
+            // CRITICAL: The size of this struct is the size of a TimeSpan (which itself is the size of a long).  This
+            // allows stopwatches to be atomically overwritten, without a concern for torn writes, as long as we're
+            // running on 64bit machines.  Make sure this value doesn't change as that will cause these current
+            // consumers to be invalid.
+            RoslynDebug.Assert(Marshal.SizeOf(typeof(SharedStopwatch)) == 8);
```

Comment:
I'd worry about this being true for VS for Mac and VS Code since we may still run in 32-bit processes, I'd imagine?

Reply:
The size will be teh same on all of those platforms.  SharedStopwatch will still be 64bits.  It's just that this is not *safe* on a 32bit machine.  But that lack of safety *already* existed with the prior code.  Torn writes were possible with the DateTime approach even prior to my change.  

The only difference now is that we're stating that at least on 64bit runtimes it's safe to use a SharedStopWatch without tearing.  on 32bits, you'll have to decide what the right thing is to do.

Reply:
i can add a comment in SolutionAssetCache.Update that says that this code may not be safe on 32bit machines.